market/mysterium/mysterium_api.go

```diff
@@ -235,12 +238,28 @@ func (mApi *MysteriumAPI) QueryProposals(query ProposalsQuery) ([]market.Service
 	if err != nil {
 		return nil, err
 	}
+	req.Header.Add("If-None-Match", mApi.latestProposalsEtag)
 
-	var proposalsResponse ProposalsResponse
-	err = mApi.http.DoRequestAndParseResponse(req, &proposalsResponse)
+	res, err := mApi.http.Do(req)
 	if err != nil {
 		return nil, err
 	}
+	defer res.Body.Close()
+
+	if res.StatusCode == http.StatusNotModified {
+		return nil, errors.New("node already contains newest proposals")
+	}
```

Comment:
This doesn't look like an error case to me. It should simply encapsulate the Not Modified case, but to the outside, it would appear as a simple proposal list.